core/except.cpp

```diff
@@ -315,7 +312,22 @@ void validSlice(const Dimensions &dims, const Slice &slice) {
                               dims[slice.dim]) ||
       slice.end > dims[slice.dim])
     throw except::SliceError("Expected " + to_string(slice) + " to be in " +
-                             to_string(dims));
+                             to_string(dims) + ".");
+}
+
+void coordsAndLabelsMatch(const DataConstProxy &a, const DataConstProxy &b) {
+  if (a.coords() != b.coords() || a.labels() != b.labels())
+    throw except::CoordMismatchError("Expected coords and labels to match.");
```

Comment:
Would it be useful to separate the `coords` and `labels` check so that we can print information regarding which was the check that failed? Or would this be of very little use?

Reply:
Yes that would probably make sense. I'd first like to see how using labels in practice works though.